ramp/post_process/post_process.py

```diff
@@ -60,17 +59,618 @@ def Profile_series_plot(stoch_profiles_series):
-    if fname is not None:
-        series_frame.to_csv(os.path.join(BASE_PATH, 'results', f'output_file_{os.path.split(fname)[-1]}.csv'))
+    path_to_write = None
+    if ofname is not None:
+        path_to_write = ofname
+    else:
+        if j is not None:
+            path_to_write = os.path.join(
+                BASE_PATH, "results", "output_file_%d.csv" % (j)
+            )
+        if fname is not None:
+            path_to_write = os.path.join(
+                BASE_PATH,
+                "results",
+                f'output_file_{os.path.split(fname)[-1].replace(".", "_")}.csv',
+            )
+
+    if path_to_write is not None:
+        print(f"Writing RAMP results to {path_to_write}")
+        series_frame.to_csv(path_to_write)
+    else:
+        print("No path to a file was provided to write the results")
+
+valid_units = ('kW',"W","MW","GW","TW")
+class Run:
+
+    def __init__(self,user,unit,calendar_years):
+
+        self.user = user
+        self.unit = unit
+
+        if isinstance(calendar_years,int):
+            calendar_years = [calendar_years]
+
+        self.calendar_years = calendar_years
+
+        datatimeIndexes = [
+            pd.date_range(start=f"{year}-01-01",periods=365*60*24,freq="1min",name = "date")
+            for year in self.calendar_years
+        ]
+        self._datetimeIndex = datatimeIndexes[0]
+
+        for year,Index in enumerate(datatimeIndexes):
+            if year != 0:
+                self._datetimeIndex=self._datetimeIndex.append(Index)
+
+    @property
+    def unit(self):
+        return self._unit
+
+    @unit.setter
+    def unit(self,var):
+        if var not in valid_units:
+            raise ValueError(f"valid units are: {valid_units}")
+
+        self._unit = var
+
+
+    def _get_DatetimeIndex(self,starting_day,n_days):
+
+        request = pd.date_range(start=starting_day,periods=n_days*24*60,freq="1min")
+
+        request_intersection = self._datetimeIndex.intersection(request)
+
+        if not request.equals(request_intersection):
+            raise ValueError("not a valid date")
+
+        return request
+
+    def generate_profiles(self,starting_day,n_days,peak_time_range,columns=["Baseline"]):
+        idx = self._get_DatetimeIndex(starting_day,n_days)
+        results = {}
+        for profile in columns:
+            profiles = []
+            for prof_i in  idx.day_of_year.unique():
+
+                result = self.user.generate_aggregated_load_profile(
+                    prof_i = prof_i,
+                    day_type = 1,
+                    peak_time_range = peak_time_range
+                )
+
+                profiles.extend(result.tolist())
+
+            results[profile] = pd.Series(index=idx,data=profiles)
+
+
+        return Plot(pd.concat(results,axis=1))
+
+
+
+class LinePlot:
+    """Line plot functions for Plotly and matplotlib engine
+    """
+    def plotly(df,**layout):
```

Comment:
```suggestion
    def plotly(self, df,**layout):
```

If this is a method class, you should add `self`, same goes for many occurences